.github/workflows/docker-ci.yml

```diff
@@ -0,0 +1,628 @@
+        id: check_vars_secrets
+        run: |
+          missing_vars=()
+          missing_secrets=()
+
+          # check Variables
+          if [ -z "${{ vars.DOCKERHUB_USERNAME }}" ]; then
+            missing_vars+=("DOCKERHUB_USERNAME")
+          fi
+
+          if [ -z "${{ vars.QUAYIO_USERNAME }}" ]; then
+            missing_vars+=("QUAYIO_USERNAME")
+          fi
+
+          if [ -z "${{ vars.ACACIA_BUCKETNAME }}" ]; then
+            missing_vars+=("ACACIA_BUCKETNAME")
+          fi
+
+          # check Secrets
+          if [ -z "${{ secrets.PAT_TOKEN }}" ]; then
+            missing_secrets+=("PAT_TOKEN")
+          fi
+
+          if [ -z "${{ secrets.DOCKERHUB_TOKEN }}" ]; then
+            missing_secrets+=("DOCKERHUB_TOKEN")
+          fi
+
+          if [ -z "${{ secrets.QUAYIO_TOKEN }}" ]; then
+            missing_secrets+=("QUAYIO_TOKEN")
+          fi
+
+          if [ -z "${{ secrets.ACACIA_ACCESS_KEY_ID }}" ]; then
+            missing_secrets+=("ACACIA_ACCESS_KEY_ID")
+          fi
+
+          if [ -z "${{ secrets.ACACIA_SECRET_ACCESS_KEY }}" ]; then
+            missing_secrets+=("ACACIA_SECRET_ACCESS_KEY")
+          fi
+
+          # If any missing variables or secrets, exit with error
+          if [ ${#missing_vars[@]} -ne 0 ] || [ ${#missing_secrets[@]} -ne 0 ]; then
+            echo "Some required variables or secrets are not set:"
+            if [ ${#missing_vars[@]} -ne 0 ]; then
+              echo "Missing Variables: ${missing_vars[@]}"
+            fi
+            if [ ${#missing_secrets[@]} -ne 0 ]; then
+              echo "Missing Secrets: ${missing_secrets[@]}"
+            fi
+            exit 1
+          else
+            echo "All required variables and secrets are set."
+          fi
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2  # Ensure enough history is available
+
+      - name: Get changed files
+        id: changed_files
+        run: |
+          files=$(git diff --name-only ${{ github.event.before }} ${{ github.sha }} '*.dockerfile')
+          echo "Files changed: $files"
+          echo "files=$files" 
+          echo "files<<EOF" >> $GITHUB_OUTPUT
+          echo "$files" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+
+
+      - name: Debug output of changed files
+        run: |
+          echo "Files from output: ${{ steps.changed_files.outputs.files }}"      
+
+
+      - name: Set proceed flag
+        id: set_proceed_flag
+        run: |
+          changed_files="${{ steps.changed_files.outputs.files }}"
+          
+          # Count files of modified Dockerfile 
+          file_count=$(echo "$changed_files" | wc -l)
+          if [ "$file_count" -eq 0 ]; then
+            echo "No Dockerfile has been modified. Setting proceed_valid to false."
+            echo "proceed_valid=false" >> $GITHUB_OUTPUT
+            exit 0
+          elif [ "$file_count" -gt 1 ]; then
```

Comment:
Let's try considering some extra logic here. The message may be enough but I think we should try expanding this out. There is also a question of the limit. If two files change, could we look into try to build two recipes?

Reply:
No, the process involves automated scanning, uploading, and downloading with automatic deployment, and they do not support simultaneous multi-file operations. Additionally, multi-file operations require careful consideration of single-file blocking issues. If multi-file support needs to be extended, I would prefer using Redis—a memory database on the runner side.